bugsnag-android-performance/src/main/kotlin/com/bugsnag/android/performance/HasAttributes.kt

```diff
@@ -52,13 +52,37 @@ public interface HasAttributes {
      * @param name the attribute name
      * @param value the value to set the attribute to
      */
-    public fun setAttribute(name: String, value: Collection<String>?)
+    public fun setAttribute(name: String, value: Array<String>?)
 
     /**
-     * Set or clear a string collection attribute. Passing `null` will remove the attribute.
+     * Set a collection of values.
      *
      * @param name the attribute name
      * @param value the value to set the attribute to
      */
-    public fun setAttribute(name: String, value: Array<String>?)
+    public fun setAttribute(name: String, value: Collection<Any>?)
```

Comment:
I think it's worth documenting how these collections need to be structured. Something along the lines of:
> Set a collection of primitive values as an attribute. Every element should be of the same type, and only `String`, `Integer`, `Long`, `Double` and `Boolean` are supported. If the collection contains mixed values, `null`, or any other type: the behavior is undefined.

@tomlongridge thoughts?

Reply:
Is it clear from the Otel spec that an array with some `null`s isn't allowed? I'd imagined a part-filled array would be acceptable and more common than actual mixed types.

Reply:
The [spec](https://opentelemetry.io/docs/specs/otel/common/) says:

> Attribute values of null are not valid and attempting to set a null value is undefined behavior.
> null values SHOULD NOT be allowed in arrays.

Reply:
Pretty clear then! Wording looks good to me.